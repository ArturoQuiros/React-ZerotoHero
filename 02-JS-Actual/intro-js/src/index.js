//console.log("Hola Mundo");
//variables
//const nombre = "Arturo";
//let apellido = "Quiros";

/* 
-----------------------------------
DES OBJ
-----------------------------------
const persona = {
  name: "Jose",
  age: 22,
  mail: "email@example.com",
};

const useContext = ({ name, age, mail, phone = "8888" }) => {
  return {
    name: name,
  };
};

const man = useContext(persona);
console.log(man);
*/

/* 
-----------------------------------
DES ARR
-----------------------------------
*/
//estandar way
const people = ["Uno", "Dos", "Tres"];
console.log(people[0]);

//desestrucutred way
const [] = people;
const [P1] = people;
const [, , P3] = people; //ignoring indexes
console.log(P1); //index 01
console.log(P3);

const returnArra = () => {
  return ["ABC", 123];
};

const [letras, numeros] = returnArra();
console.log(letras, numeros); //got the array data on names

//Tarea
const useState = (value) => {
  return (
    value,
    () => {
      console.log("Hola");
    }
  );
};

const arr = useState("Goku");
arr[1](); //calling de function

//tarea
