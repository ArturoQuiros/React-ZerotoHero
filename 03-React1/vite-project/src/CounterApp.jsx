import { useState } from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {
    
    const [ counter, setCounter ] = useState( value );

    const handleAdd = () => { 
        // console.log(event)
        setCounter( counter + 1 );
        // setCounter( (c) => c + 1 )
    }

    const handleSubstract = () => setCounter( counter - 1 );
    const handleReset = () => setCounter( value );

    return (
        <>
            <h1>CounterApp</h1>
            <h2> { counter } </h2>

            <button onClick={ handleAdd }> +1 </button>
            <button onClick={ handleSubstract }> -1 </button>
            <button aria-label="btn-reset" onClick={ handleReset }> Reset </button>
        </>
    );
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}


