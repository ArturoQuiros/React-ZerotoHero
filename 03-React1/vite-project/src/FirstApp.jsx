import PropTypes from 'prop-types';

export const FirstApp = ({ title, subTitle, name }) => {

  // console.log(props);
  
  return (
    <>
      <h1 data-testid="test-title"> { title } </h1>
      {/* <code>{ JSON.stringify( newMessage ) }</code> */}
      <p>{ subTitle }</p>
      <p>{ subTitle }</p>
      <p>{ name }</p>
    </>
  )
}


FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
}

FirstApp.defaultProps = {
  name: 'Fernando Herrera',
  subTitle: 'No hay subtítulo',
  // title: 'No hay título',
}
