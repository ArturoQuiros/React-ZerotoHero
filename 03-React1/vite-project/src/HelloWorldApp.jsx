// function App() {
//     return (<h1>Hola Mundo</h1>);
// }

// export default App;

export const HelloWorldApp = () => {


  return (
    <h1>Hello World App</h1>
  );
}

// FirstApp  <h1>Fernando</h1>
