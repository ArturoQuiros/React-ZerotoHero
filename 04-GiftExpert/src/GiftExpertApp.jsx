import { useState } from "react";
import AddCategory from "./components/AddCategory";
import GifGrid from "./components/GifGrid";

const GiftExpertApp = () => {
  const [categories, setCategories] = useState(["DBZ"]);
  const onAddCategory = (newCategorie) => {
    if (categories.includes(newCategorie)) return;
    setCategories([newCategorie, ...categories]);
  };
  return (
    <>
      <h1>Gift Expert App</h1>

      <AddCategory onNewCategory={onAddCategory} />

      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </>
  );
};

export default GiftExpertApp;
