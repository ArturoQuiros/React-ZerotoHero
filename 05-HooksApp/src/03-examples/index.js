

export * from './LoadingQuote';
export * from './MultipleCustomHooks';
export * from './Quote';