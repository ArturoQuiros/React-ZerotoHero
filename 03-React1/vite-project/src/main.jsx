import React from 'react';
import ReactDOM from 'react-dom/client';
import { CounterApp } from './CounterApp';

import { FirstApp } from './FirstApp';
// import { HelloWorldApp } from './HelloWorldApp';
import './styles.css';



ReactDOM.createRoot( document.getElementById('root') ).render(
    <React.StrictMode>
        <CounterApp value={ 20 } />
        {/* <FirstApp title="Hola, Soy Vegeta" /> */}
    </React.StrictMode>
);


