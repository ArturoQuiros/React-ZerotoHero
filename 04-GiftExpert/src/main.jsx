import React from "react";
import ReactDOM from "react-dom/client";
import GiftExpertApp from "./GiftExpertApp";
import "./styles.css";

ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    <GiftExpertApp />
  </React.StrictMode>
);
