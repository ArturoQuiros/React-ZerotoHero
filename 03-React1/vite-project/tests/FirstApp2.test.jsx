import { render, screen } from '@testing-library/react';
import { FirstApp } from '../src/FirstApp';


describe('Pruebas en <FirstApp />', () => {

    const title    = 'Hola, Soy Goku';
    const subTitle = 'Soy un subtitulo';
    
    test('debe de hacer match con el snapshot', () => {
        
        const { container } = render( <FirstApp title={ title } /> );
        expect( container ).toMatchSnapshot();

    });

    test('debe de mostrar el mensaje "Hola, Soy Goku"', () => {
        
        render( <FirstApp title={ title } /> );
        expect( screen.getByText(title) ).toBeTruthy();
        // screen.debug();
    });

    test('debe de mostrar el titulo en un h1', () => {
        render( <FirstApp title={ title } /> );
        expect( screen.getByRole('heading', { level: 1 }).innerHTML ).toContain( title );
    });

    test('debe de mostrar el subtitulo enviado por props', () => {
        
        render( 
            <FirstApp 
                title={ title }
                subTitle={ subTitle }
            />  
        );

        expect( screen.getAllByText(subTitle).length ).toBe(2);

    });


});