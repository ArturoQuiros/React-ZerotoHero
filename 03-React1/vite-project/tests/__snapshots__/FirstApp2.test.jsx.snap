// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola, Soy Goku
     
  </h1>
  <p>
    No hay subtítulo
  </p>
  <p>
    No hay subtítulo
  </p>
  <p>
    Fernando Herrera
  </p>
</div>
`;
